File: go-client-sdk/pkg/models/operations/getsubscriptionssubscriptionid.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetSubscriptionsSubscriptionIDPathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscription_id"`
}

type GetSubscriptionsSubscriptionIDRequest struct {
	PathParams GetSubscriptionsSubscriptionIDPathParams
}

type GetSubscriptionsSubscriptionIDResponse struct {
	ContentType  string
	StatusCode   int
	Subscription *shared.Subscription
}

File: typescript-client-sdk/src/sdk/models/operations/getsubscriptionssubscriptionid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetSubscriptionsSubscriptionIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscription_id" })
  subscriptionId: string;
}

export class GetSubscriptionsSubscriptionIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetSubscriptionsSubscriptionIdPathParams;
}

export class GetSubscriptionsSubscriptionIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  subscription?: shared.Subscription;
}